generic-crd-plugin/pkg/syncer/back_syncer.go

```diff
@@ -0,0 +1,298 @@
+package syncer
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/loft-sh/vcluster-generic-crd-plugin/pkg/config"
+	"github.com/loft-sh/vcluster-generic-crd-plugin/pkg/namecache"
+	"github.com/loft-sh/vcluster-generic-crd-plugin/pkg/patches"
+	"github.com/loft-sh/vcluster-sdk/syncer"
+	synccontext "github.com/loft-sh/vcluster-sdk/syncer/context"
+	"github.com/loft-sh/vcluster-sdk/syncer/translator"
+	"github.com/loft-sh/vcluster-sdk/translate"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/workqueue"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const (
+	IndexByVirtualName = "indexbyvirtualname"
+)
+
+func CreateBackSyncer(ctx *synccontext.RegisterContext, config *config.SyncBack, parentConfig *config.FromVirtualCluster, parentNC namecache.NameCache) (syncer.Syncer, error) {
+	if len(config.Selectors) == 0 {
+		return nil, fmt.Errorf("the syncBack config for %s (%s) is missing Selectors", config.Kind, parentConfig.Kind)
+	}
+
+	obj := &unstructured.Unstructured{}
+	obj.SetKind(config.Kind)
+	obj.SetAPIVersion(config.ApiVersion)
+
+	// TODO: [low priority] check if config.Kind + config.ApiVersion has status subresource
+	statusIsSubresource := true
+
+	//  |
+	//  |
+	//  |
+	//  |
+	// \|/
+	// TODO: TOP priority - initialize a name cache for the translations done in the syncBack patches and reversePatches
+	nc := parentNC
```

Comment:
This is next step number 2. The syncback controller probably should not use the same cache as the parent syncer.